pkg/server/test/list_objects.go

```diff
@@ -14,460 +12,205 @@ import (
-		}
-
-		listObjectsQuery := &commands.ListObjectsQuery{
-			Datastore:             ds,
-			Logger:                logger.NewNoopLogger(),
-			ListObjectsDeadline:   defaultListObjectsDeadline,
-			ListObjectsMaxResults: defaultListObjectsMaxResults,
-			ResolveNodeLimit:      defaultResolveNodeLimit,
-			ConnectedObjects:      connectedObjectsCmd.StreamedConnectedObjects,
-			CheckResolver:         graph.NewLocalChecker(storage.NewContextualTupleDatastore(ds), checkConcurrencyLimit),
-		}
-
-		ctx = typesystem.ContextWithTypesystem(ctx, typesys)
-		runListObjectsTests(t, ctx, testCases, listObjectsQuery)
-	})
-}
-
 type mockStreamServer struct {
 	grpc.ServerStream
 	channel chan string
 }
 
-func NewMockStreamServer(size int) *mockStreamServer {
-	return &mockStreamServer{
-		channel: make(chan string, size),
-	}
-}
-
 func (x *mockStreamServer) Send(m *openfgapb.StreamedListObjectsResponse) error {
 	x.channel <- m.Object
 	return nil
 }
 
-func runListObjectsTests(t *testing.T, ctx context.Context, testCases []listObjectsTestCase, listObjectsQuery *commands.ListObjectsQuery) {
-
-	for _, test := range testCases {
-		ctx = storage.ContextWithContextualTuples(ctx, test.request.ContextualTuples.GetTupleKeys())
-
-		t.Run(test.name+"/streaming", func(t *testing.T) {
-			server := NewMockStreamServer(len(test.expectedResult))
-
-			done := make(chan struct{})
-			var streamedObjectIds []string
-			go func() {
-				for x := range server.channel {
-					streamedObjectIds = append(streamedObjectIds, x)
-				}
-
-				done <- struct{}{}
-			}()
-
-			err := listObjectsQuery.ExecuteStreamed(ctx, &openfgapb.StreamedListObjectsRequest{
-				StoreId:              test.request.StoreId,
-				AuthorizationModelId: test.request.AuthorizationModelId,
-				Type:                 test.request.Type,
-				Relation:             test.request.Relation,
-				User:                 test.request.User,
-				ContextualTuples:     test.request.ContextualTuples,
-			}, server)
-
-			close(server.channel)
-			<-done
+type listObjectsTestCase struct {
+	name                   string
+	schema                 string
+	tuples                 []*openfgapb.TupleKey
+	model                  string
+	objectType             string
+	user                   string
+	relation               string
+	contextualTuples       *openfgapb.ContextualTupleKeys
+	allResults             []string //all the results. the server may return less
+	maxResults             uint32
+	minimumResultsExpected uint32
+}
 
-			require.ErrorIs(t, err, test.expectedError)
-			require.LessOrEqual(t, len(streamedObjectIds), defaultListObjectsMaxResults)
-			require.Subset(t, test.expectedResult, streamedObjectIds)
-		})
+func TestListObjectsRespectsMaxResults(t *testing.T, ds storage.OpenFGADatastore) {
+	testCases := []listObjectsTestCase{
+		{
+			name:   "respects_when_schema_1_1_and_reverse_expansion_implementation",
+			schema: typesystem.SchemaVersion1_1,
+			model: `
+				type user
```

Comment:
Indentation is excessive here. Please fix the whitespace. You can look at the spacing in the other test files that include models to get it dialed in.